Repositories/ProductRepository.cs

```diff
@@ -15,33 +15,33 @@ public ProductRepository(ProductContext context)
         {
             _context = context;
         }
-        public async Task<List<Product>> GetAll() 
+        public async Task<List<Product>> GetAllAsync() 
         {
             return await _context.Products.ToListAsync();
         }
 
-        public async Task<Product> GetById(int id) => await _context.Products.FirstOrDefaultAsync(p => p.Id == id);
-        public async Task DeleteById(int id) 
+        public async Task<Product> GetByIdAsync(int id) => await _context.Products.FirstOrDefaultAsync(p => p.Id == id);
+        public async Task DeleteByIdAsync(int id) 
         {
             var itemToBeDeleted = await _context.Products.FirstOrDefaultAsync(p => p.Id == id);
             if (itemToBeDeleted != null)
                 _context.Products.Remove(itemToBeDeleted);
 
-            await SaveChanges();    
+            await SaveChangesAsync();    
         }
 
-        public async Task Add(Product product) 
+        public async Task AddAsync(Product product) 
         {
             await _context.Products.AddAsync(product);
-            await SaveChanges();
+            await SaveChangesAsync();
         }
 
-        public async Task SaveChanges()
+        public async Task SaveChangesAsync()
```

Comment:
What is the reason to use SaveChangesAsync as a separate repository and interface method when you use it in every other repo method and don't use in services?